src/commands/emojistats.rs

```diff
@@ -2,25 +2,27 @@ use std::collections::hash_map::HashMap;
 
 use super::*;
 
-use crate::db::emoji_logging::EmojiStats;
+use crate::db::emoji_logging::{EmojiStats, Ordering};
 
 #[command]
-#[usage("emojistats [in_text or reactions]")]
-#[usage("emojistats [emoji] ")]
-#[usage("emojistats [emoji] [in_text or reactions]")]
+#[usage("emojistats [emoji] | --least")]
 pub async fn emojistats(ctx: &client::Context, msg: &Message, mut args: Args) -> CommandResult {
     let db = ctx.get_db().await;
 
-    let single_emoji = match args.single_quoted::<String>().ok() {
+    let (ordering, single_emoji) = match args.single_quoted::<String>().ok().as_deref() {
+        Some("--least") => (Some(Ordering::Ascending), None),
         Some(arg) => {
             let found_emoji = match crate::util::find_emojis(&arg).first() {
                 Some(searched_emoji) => db.get_emoji_usage_by_id(searched_emoji).await,
                 None => db.get_emoji_usage_by_name(&arg).await,
             };
 
-            Some(found_emoji.user_error("Could not find that emote")?)
+            (
+                None,
+                Some(found_emoji.user_error("Could not find that emote")?),
+            )
         }
-        None => None,
+        None => (Some(Ordering::Descending), None),
     };
```

Comment:
Why not just have ordering be Descending by default, even in the case where a specific emote is searched? While Ordering then does not "do" anything, it also doesn't hurt - and it remove the case where there is no ordering set (which is not actually gonna happen in the case of listing all emoji, but rust doesn't know that).

also, small style thing, i'd put the None and the Some("--least") case next to each other (at the top), and have the Some(arg) case be the last, as it semantically is more different from the other two

Reply:
I literally was like do exactly that. but mzlvk was like nahh, you can check my old commits

Reply:
It is descending by default

Reply:
Wdym remove hte case when its not set?